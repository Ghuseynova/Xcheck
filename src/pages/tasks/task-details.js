import React from 'react';
import { PageHeader, Button } from 'antd';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import TaskDescription from '../../components/tasks/task-description';

export const TaskDetails = (props) => {
  const { tasks } = props;
  return (
    <div className="wrapper">
      <PageHeader className="site-page-header" title="Task details" />
      <Button type="primary">
        <Link to="/tasks/">Back</Link>
      </Button>
      {tasks?.formValues && tasks.formValues[0] && <TaskDescription task={tasks.formValues[0]} />}
    </div>
  );
};

TaskDetails.propTypes = {
  tasks: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

const mapStateToProps = ({ tasks }) => ({
  tasks,
});

export default connect(mapStateToProps)(TaskDetails);
