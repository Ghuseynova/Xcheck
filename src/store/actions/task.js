import {
  GET_TASK_SESSION_BY_ID,
  POST_TASK_SESSIONS,
  UPDATE_TASK_SESSION,
  GET_TASKSTABLE_SESSIONS,
} from './types/task';

const getTaskSessionById = (id) => {
  return {
    type: GET_TASK_SESSION_BY_ID,
    payload: id,
  };
};

const postTaskSession = (values) => {
  return {
    type: POST_TASK_SESSIONS,
    payload: values,
  };
};

const updateTaskSession = (values) => {
  return {
    type: UPDATE_TASK_SESSION,
    payload: values,
  };
};

const getTasksTable = () => {
  return {
    type: GET_TASKSTABLE_SESSIONS,
  };
};

export { getTaskSessionById, postTaskSession, updateTaskSession, getTasksTable };
