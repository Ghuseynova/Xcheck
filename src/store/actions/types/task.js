const GET_TASKSTABLE_SESSIONS = 'GET_TASKSTABLE_SESSIONS';
const GET_TASKSTABLE_SESSIONS_SUCCESS = 'GET_TASKSTABLE_SESSIONS_SUCCESS';
const GET_TASKSTABLE_SESSIONS_FAILURE = 'GET_TASKSTABLE_SESSIONS_FAILURE';

const GET_TASK = 'GET_TASK';
const GET_TASK_SUCCESS = 'GET_TASK_SUCCESS';
const GET_TASK_FAILURE = 'GET_TASK_FAILURE';

const UPDATE_TASK_SESSION = 'UPDATE_TASK_SESSION';
const UPDATE_TASK_SESSION_SUCCESS = 'UPDATE_TASK_SESSION_SUCCESS';
const UPDATE_TASK_SESSION_FAILURE = 'UPDATE_TASK_SESSION_FAILURE';

const POST_TASK_SESSIONS = 'POST_TASK_SESSIONS';
const POST_TASK_SESSIONS_SUCCESS = 'POST_TASK_SESSIONS_SUCCESS';
const POST_TASK_SESSIONS_FAILURE = 'POST_TASK_SESSIONS_FAILURE';

export {
  GET_TASKSTABLE_SESSIONS,
  GET_TASKSTABLE_SESSIONS_SUCCESS,
  GET_TASKSTABLE_SESSIONS_FAILURE,
  POST_TASK_SESSIONS,
  POST_TASK_SESSIONS_SUCCESS,
  POST_TASK_SESSIONS_FAILURE,
  GET_TASK,
  GET_TASK_SUCCESS,
  GET_TASK_FAILURE,
  UPDATE_TASK_SESSION,
  UPDATE_TASK_SESSION_SUCCESS,
  UPDATE_TASK_SESSION_FAILURE,
};
