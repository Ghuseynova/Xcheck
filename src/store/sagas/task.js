import { takeEvery, call, put } from 'redux-saga/effects';
import { axiosDB } from '../../axios';
import {
  POST_TASK_SESSIONS,
  POST_TASK_SESSIONS_SUCCESS,
  POST_TASK_SESSIONS_FAILURE,
  UPDATE_TASK_SESSION_SUCCESS,
  UPDATE_TASK_SESSION_FAILURE,
  UPDATE_TASK_SESSION,
  GET_TASKSTABLE_SESSIONS,
  GET_TASKSTABLE_SESSIONS_SUCCESS,
  GET_TASKSTABLE_SESSIONS_FAILURE,
  GET_TASK_SESSION_BY_ID,
} from '../actions/types/task';

function* workerPostTask(action) {
  const uri = 'http://localhost:3000/tasks';
  try {
    yield axiosDB.post('tasks', action.payload);
    yield put({ type: POST_TASK_SESSIONS_SUCCESS });
  } catch {
    yield put({
      type: POST_TASK_SESSIONS_FAILURE,
      payload: `ERROR! Cannot post task at ${uri}`,
    });
  }
}

function* workerUpdateTaskSession(action) {
  try {
    yield axiosDB.put(`/tasks/${action.payload.id}`, action.payload);
    yield put({ type: UPDATE_TASK_SESSION_SUCCESS });
  } catch {
    yield put({
      type: UPDATE_TASK_SESSION_FAILURE,
      payload: `ERROR! Cannot update task session with this ID`,
    });
  }
}

function* workerGetTasksTable() {
  const uri = 'http://localhost:3000/tasks';
  try {
    const result = yield axiosDB.get('tasks');
    yield put({ type: GET_TASKSTABLE_SESSIONS_SUCCESS, payload: result.data });
  } catch {
    yield put({
      type: GET_TASKSTABLE_SESSIONS_FAILURE,
      payload: `ERROR! Cannot get tasks at ${uri}`,
    });
  }
}

function* workerGetTaskTableById(action) {
  const uri = `http://localhost:3000/tasks?taskId=${action.payload}`;
  try {
    const result = yield axiosDB.get(`/tasks?taskId=${action.payload}`);
    yield put({ type: GET_TASKSTABLE_SESSIONS_SUCCESS, payload: result.data });
  } catch {
    yield put({
      type: GET_TASKSTABLE_SESSIONS_FAILURE,
      payload: `ERROR! Cannot get task at ${uri}`,
    });
  }
}

function* watchTask() {
  yield takeEvery(POST_TASK_SESSIONS, workerPostTask);
  yield takeEvery(UPDATE_TASK_SESSION, workerUpdateTaskSession);
  yield takeEvery(GET_TASKSTABLE_SESSIONS, workerGetTasksTable);
  yield takeEvery(GET_TASK_SESSION_BY_ID, workerGetTaskTableById);
}

export default watchTask;
