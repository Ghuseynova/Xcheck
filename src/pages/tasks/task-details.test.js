import React from 'react';
import { mount } from 'enzyme';
import { Router } from "react-router-dom";
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import configureStore from 'redux-mock-store';
import { TaskDetails } from './task-details';
import task from './test-data';

describe('TaskDetails page', () => {
	const mockStore = configureStore();
	let store;
	const defaultProps = {
		tasks: task
	}
	beforeEach(() => {
		store = mockStore({ task });
	});

	it('toMatchSnapshot TaskDetails page', () => {
		const tree = mount(<Router history={createMemoryHistory()}><TaskDetails {...defaultProps} /></Router>);
		expect(tree).toMatchSnapshot();
	});

	it('Should render header, button for TaskDetails page', () => {
		const wrapper = mount(
			<Router history={createMemoryHistory()}>
				<Provider store={store}>
					<TaskDetails {...defaultProps} />
				</Provider>
			</Router>,
		)
		const header = wrapper.find('.ant-page-header-heading-title');
		const buttonAdd = wrapper.find('.ant-btn-primary');
		expect(header.length).toBe(1);
		expect(buttonAdd.length).toBe(1);
	});
});