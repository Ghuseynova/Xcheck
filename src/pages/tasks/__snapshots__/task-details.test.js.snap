// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetails page toMatchSnapshot TaskDetails page 1`] = `ReactWrapper {}`;
